src/app/components/DynamicContentStyles.tsx

```diff
@@ -12,37 +16,51 @@ export const WithStyles = styled.div`
 const cacheStyles = new Map<string, string>();
 
 interface DynamicContentStylesProps extends React.HTMLAttributes<HTMLDivElement> {
+  book: State['book'];
   disable?: boolean;
 }
 
 // tslint:disable-next-line: variable-name
 const DynamicContentStyles = React.forwardRef<HTMLElement, DynamicContentStylesProps>((
-  { children, disable, ...otherProps }: React.PropsWithChildren<DynamicContentStylesProps>,
+  { book, children, disable, ...otherProps }: React.PropsWithChildren<DynamicContentStylesProps>,
   ref
 ) => {
-  const queryParams = useSelector(query);
   const [styles, setStyles] = React.useState('');
+  const { archiveLoader } = useServices();
+  const bookConfig = book && getBookVersionFromUUIDSync(book.id);
+  const queryParams = useSelector(query);
 
   React.useEffect(() => {
     if (disable) {
       setStyles('');
       return;
     }
 
-    const cssfileUrl = queryParams['content-style'];
+    let cssfileUrl = queryParams['content-style'];
+
+    if (!cssfileUrl && bookConfig?.dynamicStyles && book?.style_href) {
+      cssfileUrl = book.style_href;
+
+      if (!isAbsoluteUrl(cssfileUrl)) {
+        const contentUrl = archiveLoader.book(book.id, book.version).url();
+        cssfileUrl = fromRelativeUrl(contentUrl, cssfileUrl);
+      }
+    }
+
     if (cssfileUrl && typeof cssfileUrl === 'string') {
       if (cacheStyles.has(cssfileUrl)) {
         setStyles(assertDefined(cacheStyles.get(cssfileUrl), `we've just checked for this`));
       } else {
         fetch(cssfileUrl)
           .then((res) => res.text())
           .then((data) => {
-            cacheStyles.set(cssfileUrl, data);
+            // we are inside an "if" that checks that cssfileUrl is defined and a string
```

Comment:
it doesn't trust you because you initialized it with `let` and there might be async logic that changes it. if you change your initialization to use `const` it'll stop complaining. if you don't want to re-write your logic you can move it to a function so that you can use a return value in the `const cssfileUrl = figureOutTheCssUrl()` line

Reply:
Alright, I moved the cssFileUrl logic to a separate function